paginator/option.go

```diff
@@ -1,9 +1,17 @@
 package paginator
 
+type Flag string
+
+const (
+	ENABLED  Flag = "ENABLED"
+	DISABLED Flag = "DISABLED"
+)
+
 var defaultConfig = Config{
-	Keys:  []string{"ID"},
-	Limit: 10,
-	Order: DESC,
+	Keys:     []string{"ID"},
+	Limit:    10,
+	Order:    DESC,
+	TupleCmp: DISABLED,
```

Comment:
What about just using bool for this flag, and default to `false`?

Reply:
And `AllowTupleCmp` would be a better name to me, which is also aligning with the internal state.

Reply:
We cannot use boolean directly here. Because of the way the configuration code is structured right now, each option needs to have some value that indicates the "not set, please ignore" state. For strings, we (ab)use `""` for this purpose, and for integers, `0` is treated as unset marker. But since booleans only have two valid values that both carry meaning for us, we cannot use them directly.

This is why I opted to implement a new flag type that gives us that additional state. Alternatives would be:

1. Use pointer to a boolean, which will give us `nil` as an additional state, but will make usage awkward (`&false` is not something we can do in go, so we would either need a monstrosity like `&[]bool{false}[0]` or a helper function).
2. Do not allow resetting the tuple comparison option and treat `false` as an unset state option.

It would be ideal if Go's stdlib would provide an optional type that we could use here, but we are not there yet. And while I could write an `Option[T]` wrapper and use it here, that would stick out like a sore thumb from the rest of the code.

That being said, I am open to suggestions on how to proceed here.

Reply:
I agree with that pointer is bothering to use. 😔 Although the second alternative is more favorable to me, current options are all implemented based on a shared config structure, which might accidentally override the state if we treat `false` as unset. For instance:

```go
paginator.New(
    WithAllowTupleCmp(true),
    // AllowTupleCmp would be overridden to false in other option
    WithRules(...)
)
```
Unless with further refinement of config and option, `Flag` would be the best choice to this feature. We can go for `Flag` at this moment. 🙌

What about the naming? In my opinion naming the flag to `AllowTupleCmp` would be more consistent across the whole codebase. 🙂

Reply:
> What about the naming?

Sorry, I forgot to comment on that and update the code. I am OK with rename.

Reply:
That's fine. After document refinement, this PR is ready to go 🚀